shroud/generate.py

```diff
@@ -277,6 +279,24 @@ def check_arg_attrs(self, node, arg):
             # default to 1-d assumed shape
             attrs["dimension"] = ":"
 
+        # charlen
+        # Only meaningful with 'char *arg+intent(out)'
+        # XXX - Python needs a value if 'char *+intent(out)'
```

Comment:
What does `XXX` mean here?

Reply:
That's 'note to self'.  I added this attribute to fix a problem someone had with wrapping Python.  This function should do some better error checking to report if the attribute is missing.

Reply:
Got it -- that's what I thought, but it looks like you have a workaround (i.e. setting `charlen` to `None`)

Reply:
If the user forgets to set the attribute, they'll get a wrapper which compiles which will segfault since it will overwrite memory.